partiql-eval/src/plan.rs

```diff
@@ -334,7 +371,10 @@ impl EvaluatorPlanner {
                 match pattern {
                     Pattern::Like(logical::LikeMatch { pattern, escape }) => {
                         // TODO statically assert escape length
-                        assert!(escape.chars().count() <= 1);
+                        if escape.chars().count() > 1 {
+                            self.errors
+                                .push(EvaluationError::InvalidLikeEscape(escape.to_string()));
```

Comment:
I'm not sure this is worth its own error variant. Perhaps just use `IllegalState` like coalesce